servicemesh-install/topics/appendix-threescale-adapter.adoc

```diff
@@ -0,0 +1,146 @@
+[[threescale-adapter-tutorial]]
+= Appendix: 3scale Istio Adapter tutorial
+The 3scale Istio Adapter allows you to label a service running within the {ProductName} and integrate that service with the 3Scale API Management solution.
+
+Prerequisites:
+
+* {ProductName} 0.7.0+
+* A working 3scale account (https://www.3scale.net/signup/[SaaS] or https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.4/html/infrastructure/onpremises-installation[On-Premises])
+* https://docs.openshift.com/container-platform/3.11/servicemesh-install/servicemesh-install.html#service-mesh-install_prerequisites[{ProductName} prerequisistes]
+
+[NOTE]
+====
+To configure the 3scale Istio Adapter, refer to https://docs.openshift.com/container-platform/3.11/servicemesh-install/servicemesh-install.html#install_chapter_3[Installing Service Mesh] for instructions on adding adapter parameters to the custom resource file.
+====
+
+== Integrate the adapter with {ProductName}
+You can use these samples to configure request to your services using the 3scale Istio Adapter.
+
+[NOTE]
+====
+Pay particular attention to the `kind: handler` resource. You must update this with your 3scale credentials and the service ID of the API you want to manage.
+====
+
+Modify the handler configuration with your 3scale configuration.
+
+.Handler for the threescale adapter
+----
+apiVersion: "config.istio.io/v1alpha2"
+kind: handler
+metadata:
+  name: threescale
+  namespace: istio-system
+spec:
+  adapter: threescale
+  params:
+    service_id: "123456"
+    system_url: "http://127.0.0.1:8090"
+    access_token: "secret-token"
+  connection:
+    address: "threescale-istio-adapter:3333"
+----
+
+.Instance for template authorization
+----
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: threescale-authorization
+  namespace: istio-system
+spec:
+  template: threescale-authorization
+  params:
+    action:
+      path: request.path | "/"
+      method: request.method | "get"
+----
+
+.Rule to dispatch to threescale handler
+----
+apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: threescale
+  namespace: istio-system
+spec:
+  match: destination.labels["service-mesh.3scale.net"] == "true"
+  actions:
+  - handler: threescale.handler.istio-system
+    instances:
+    - threescale-authorization
+----
+
+=== Routing service traffic through the adapter
+In order to drive traffic for your service through the adapter and be managed by 3scale, we need to match the rule `destination.labels["service-mesh.3scale.net"] == "true"` we previously created in the configuration, in the kind: rule resource.
+
+To do so, we need to add a label to the PodTemplateSpec on the Deployment of the target workload. For example, if we have a `productpage` service, whose Pod is managed by the `productpage-v1` deployment, by adding the above label under `spec.template.labels` in `productpage-v1`, we can have the adapter authorize requests to this service.
```

Comment:
To do so, you must add a label to the PodTemplateSpec on the Deployment of the target workload. For example, if you have a `productpage` service, whose Pod is managed by the `productpage-v1` deployment, by adding the above label under `spec.template.labels` in `productpage-v1`, you can have the adapter authorize requests to this service.

(Replace "we" with "you")